futaba/cogs/moderation/manual_mod_action_warn.py

```diff
@@ -0,0 +1,179 @@
+#
+# cogs/moderation/manual_mod_action_warn.py
+#
+# futaba - A Discord Mod bot for the Programming server
+# Copyright (c) 2017-2018 Jake Richardson, Ammon Smith, jackylam5
+#
+# futaba is available free of charge under the terms of the MIT
+# License. You are free to redistribute and/or modify it under those
+# terms. It is distributed in the hopes that it will be useful, but
+# WITHOUT ANY WARRANTY. See the LICENSE file for more details.
+#
+
+"""
+Cog to warn when a mod action is done manually, instead of through the bot.
+"""
+import logging
+from datetime import datetime
+from enum import Enum
+
+from discord import AuditLogAction
+
+from futaba.cogs.tracker import get_removal_cause
+from futaba.enums import ManualModActionType, MemberLeaveType
+
+logger = logging.getLogger(__name__)
+
+__all__ = "ManualModActionWarn"
+
+
+manual_mod_action_command_map = {
+    ManualModActionType.SPECIAL_ROLE_MUTE: "{prefix}mute",
+    ManualModActionType.SPECIAL_ROLE_JAIL: "{prefix}jail or {prefix}dunce",
+    ManualModActionType.KICK_MEMBER: "{prefix}kick",
+    ManualModActionType.BAN_MEMBER: "{prefix}ban",
+}
+
+
+class ManualModActionWarn:
+    """
+    Warn moderators when they invoke a mod action manually.
+    """
+
+    __slots__ = ("bot",)
+
+    def __init__(self, bot):
+        self.bot = bot
+
+    async def dispatch_manual_action_warning(
+        self, guild, action, moderator, target, **kwargs
+    ):
+        """Dispatch a warning about invoking a moderation action manually."""
+
+        logger.info(
+            "Moderator '%s' (%d) completed a moderation action (%s) manually on member: '%s' (%d)",
+            moderator.name,
+            moderator.id,
+            action.name,
+            target.name,
+            target.id,
+        )
+
+        message_template = """
+ Hey, it looks like you did some moderation action manually when you could have used the bot.
+
```

Comment:
I think we should change the message. In my opinion it should be more to-the-point, such as:
```
You manually applied the "guest" role, which normally should be managed automatically.
```
or
```
You manually kicked @ghost (ghost#0000) (reason: `spammer`). In the future, use the command !kick.
```

We can't use role mentions in PMs unfortunately, but they're great for clarity in my opinion.